currying.md

```diff
@@ -1,2 +1,97 @@
+
+Now what if multiplication was a function instead of an operator? It would probably look like this:
+```typescript
+function multiply(x: number, y: number): number {
+    return x * y;
+}
+```
+
+Then if we were to partially apply `10` to the `multiply` function, it should be very similar to the `multiplyByTen` function above:
+```typescript
+function multiplyByTen(x: number): number {
+    return multiply(x, 10);
+}
+```
+
+What if we wanted to make partial application of `multiply` more convenient?
+```typescript
+function curriedMultiply(x: number): number => number {
+    return (y: number) => { return x * y }
+}
+```
+
+At first glance, you may be wondering, how is this more convenient? I can barely read it! Let's take a thorough look at it. The beginning of the function should be fairly easy to grasp: this function takes a number `x`, which is pretty normal, but what's not so normal is that it returns a function which takes another `number` and itself returns a `number`. Now take a look at the `return` inside the body of `curriedMultiply`. It takes a number `y` and multiplies it with `x`. Let's play this out by calling `curriedMultiply` with `10`.
+
+```typescript
+let multiplyByTen = curriedMultiply(10)
+```
+
+If we look inside this `multiplyByTen`, what do you think we would see? `x` is now `10`, but `y` doesn't have a value:
+```typescript
+curriedMultiply(10): number => number {
+    return (y: number) => { return 10 * y }
+}
+
+multiplyByTen === (y: number) => { return 10 * y }
+```
+
+So now `multiplyByTen` here is just like the `multiplyByTen` example above. We can call it with another value:
+```typescript
+let multiplyByTen = curriedMultiply(10)
+let result = multiplyByTen(5)
+result === 50
+```
+
+Since calling `curriedMultiply` returns a function, we can invoke that function immediately with another paretheses and input:
+```typescript
+curriedMultiply(10)(5) === 50
+```
+
+So effectively what we've done here is transform a function call from `function(a, b)` to `function(a)(b)`.
+
+This technique is called "currying". To be a little more precise, the reason for currying is so a function have only one input and only one output. If we look back at `curriedMultiply`, it only has one input (`x`) and has only one output (the function that takes `y`).
+
+With the currying technique, you can make new functions on the fly, for example:
+```typescript
+let times2 = curriedMultiply(2)
+
+times2(5) === 10
+times2(321) === 642
+
+let flipSign = curriedMultiply(-1)
+
+flipSign(7) === -7
+flipSign(-3) === 3
+```
+
+In PureScript, currying is a built-in feature. Again if we had `multiply` as a function:
+```purescript
+multiply :: Int -> Int -> Int
+multiply x y = x * y
+```
+
+We can partially apply a value to `multiply` and create new functions:
+
+```purescript
+times2 = multiply 2
+times2 5 == 10
+times2 321 == 642
+
+flipSign = multiply (-1)
+flipSign 7 === (-7)
+flipSign (-3) === 3
+```
```

Comment:
very easy to understand once I taught myself the syntax

Reply:
`very easy to understand once I taught myself the syntax`
This is a great comment because it reminds me that I need to add more resources for learning the syntax of these languages